roffit

```diff
@@ -256,6 +256,33 @@ sub linkfile {
             # convert https://, http:// and ftp:// URLs to <a href> links
             $field =~ s/(^|\W)((https|http|ftp):\/\/[a-z0-9\-._~%:\/?\#\[\]\@!\$&'()*+,;=]+)/$1<a href=\"$2\">$2<\/a>/gi;
 
+            # remove punctuation characters at end of url
+            $field =~ s/(href=\")([^\"]*)(&quot;)/$1$2/g;
+            $field =~ s/(href=\"(http:\/\/|https:\/\/|ftp:\/\/)\")*([,.;()*])*(\">)/$1$4/g;
```

Comment:
Shouldn't this rather correct the pattern on line 257 to not include punctuation  as terminating symbols?

Reply:
Tried with variations of pattern. Either my corrections were flawed or the URLs need to be built first.

Reply:
> Shouldn't this rather correct the pattern on line 257 to not include punctuation as terminating symbols?

Kept getting unexpected results like `<a href="https://curl">https://curl</a>.se/,` or inline punctuations being removed.